internal/gamepaddb/gen.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2024 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build ignore
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+const license = `// Copyright 2024 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+`
+
+// gamecontrollerdb.txt is downloaded at https://github.com/mdqinc/SDL_GameControllerDB.
+
+// To update the database file, run:
+//
+//     curl --location --remote-name https://raw.githubusercontent.com/mdqinc/SDL_GameControllerDB/master/gamecontrollerdb.txt
+
+//go:embed gamecontrollerdb.txt
+var gameControllerDB []byte
+
+const dbTemplate string = `{{.License}}
+
+{{.DoNotEdit}}
+
+{{.BuildConstraints}}
+
+package gamepaddb
+
+import (
+	_ "embed"
+)
+
+//go:embed gamecontrollerdb_{{.FileNameSuffix}}.txt
+var controllerBytes []byte
+//go:embed gamecontrollerdb_glfw.txt
+var glfwControllerBytes []byte
```

Comment:
This file is only for GLFW environments, right? We don't have to 'generate' this part. We should have `db_glfw.go` with the build constraint `//go:build !android && !ios && !js && !microsoftgdk && !nintendosdk && !playstation5`. `db_glfw.go` would be very similar to `db.go`. The current `db.go` doesn't have the constaints for android and ios, but I think we need this constraints as GLFW is only for desktops. We don't have to generate `db_glfw.go` by gen.go.